prov/gni/src/gnix_cq.c

```diff
@@ -247,6 +248,27 @@ static struct slist_entry *alloc_cq_entry(size_t size)
 	return NULL;
 }
 
+static int __gnix_cq_progress(struct gnix_fid_cq *cq)
+{
+	struct gnix_cq_poll_nic *pnic, *tmp;
+	int rc;
+
+	rwlock_rdlock(&cq->nic_lock);
+
+	dlist_for_each_safe(&cq->poll_nics, pnic, tmp, list) {
+		GNIX_INFO(FI_LOG_CQ, "progressing nic %p\n", pnic->nic);
+		rc = _gnix_nic_progress(pnic->nic);
+		if (rc) {
+			GNIX_WARN(FI_LOG_CQ,
+				  "_gnix_nic_progress failed: %d\n", rc);
+		}
+	}
+
+	rwlock_unlock(&cq->nic_lock);
+
+	return FI_SUCCESS;
+}
```

Comment:
Do you want that change in this initial version?  Is that the last of the comments?